File: figure3.py
"""
Reproduce Figure 3 from Bansak, Blanco, Coon & Dieringer (2025),
"Border Walls and Death on the US-Mexico Border":
    Figure 3. Migrant Deaths, 2000-2019 (complete data set)

Shares its base layer (basemap, fence, styling) with figure4.py and
figure5.py via basemap_common.py -- see that file for data requirements
and layer details. This file only picks which death points to plot.

To run in VS Code: see the instructions at the top of figure4.py -- same
.venv/ setup and re-exec trick apply here.
"""

import os
import sys
from pathlib import Path

_VENV_DIR = Path(__file__).resolve().parent / ".venv"
_VENV_PYTHON = _VENV_DIR / "bin" / "python"
if _VENV_PYTHON.exists() and Path(sys.prefix).resolve() != _VENV_DIR.resolve():
    os.execv(str(_VENV_PYTHON), [str(_VENV_PYTHON), str(Path(__file__).resolve()), *sys.argv[1:]])

import basemap_common as bc

deaths = bc.load_deaths()
subset = deaths[deaths["is_pre_sfa"] | deaths["is_post_sfa"]]
print(f"All deaths, 2000-2019, in this extract: {len(subset)}")
if len(subset) != 3041:
    print("NOTE: this does not match the paper's reported 3,041 -- the CSV "
          "may have been updated with additional records since the paper's "
          "data pull. Check the count above against Table 4.")

bc.render_figure(
    deaths_subset=subset,
    death_label="Location of Remains 2000-2019",
    title="Figure 3: Migrant Deaths, 2000-2019",
    out_filename="figure3_reproduction.png",
)

File: r/figure3.R
# Reproduce Figure 3 from Bansak, Blanco, Coon & Dieringer (2025),
# "Border Walls and Death on the US-Mexico Border":
#     Figure 3. Migrant Deaths, 2000-2019 (complete data set)
#
# R port of figure3.py -- shares its base layer with figure4.R/figure5.R via
# basemap_common.R. See README.md for the full pipeline description.
#
# Requires the "sf" package: install.packages("sf")
# To run (from the repo root): Rscript r/figure3.R

.get_script_dir <- function() {
  cmd_args <- commandArgs(trailingOnly = FALSE)
  file_flag <- "--file="
  m <- grep(file_flag, cmd_args)
  if (length(m) > 0) return(dirname(normalizePath(sub(file_flag, "", cmd_args[m]))))
  rp <- tryCatch({
    if (requireNamespace("rstudioapi", quietly = TRUE) && rstudioapi::isAvailable())
      rstudioapi::getActiveDocumentContext()$path else NULL
  }, error = function(e) NULL)
  if (!is.null(rp) && nzchar(rp)) return(dirname(normalizePath(rp)))
  getwd()
}
SCRIPT_DIR <- .get_script_dir()
source(file.path(SCRIPT_DIR, "basemap_common.R"))

deaths <- load_deaths()
subset <- deaths[deaths$is_pre_sfa | deaths$is_post_sfa, ]
cat(sprintf("All deaths, 2000-2019, in this extract: %d\n", nrow(subset)))
if (nrow(subset) != 3041) {
  cat("NOTE: this does not match the paper's reported 3,041 -- the CSV\n",
      "may have been updated with additional records since the paper's\n",
      "data pull. Check the count above against Table 4.\n")
}

render_figure(
  deaths_subset = subset,
  death_label = "Location of Remains 2000-2019",
  title = "Figure 3: Migrant Deaths, 2000-2019",
  out_filename = "figure3_reproduction_R.png"
)
